interactions/client/get.py

```diff
@@ -0,0 +1,398 @@
+        if isinstance(obj, _GenericAlias) or _check():
+            _obj: Type[_T] = get_args(obj)[0]
+            _objects: List[Union[_obj, Coroutine]] = []
+            kwarg_name += "s"
+
+            force_cache = kwargs.pop("force", None) == "cache"
+            force_http = kwargs.pop("force", None) == "http"
+
+            if not force_http:
+                _objects = _get_cache(_obj, client, kwarg_name, _list=True, **kwargs)
+
+            if force_cache:
+                return _objects
+
+            elif not force_http and None not in _objects:
+                return _return_cache(_objects)
+
+            elif force_http:
+                _objects.clear()
+                _func = getattr(http_name, client._http)
+                for _id in kwargs.get(kwarg_name):
+                    _kwargs = kwargs
+                    _kwargs.pop(kwarg_name)
+                    _kwargs[kwarg_name[:-1]] = _id
+                    _objects.append(_func(**_kwargs))
+                return _http_request(_obj, http=client._http, request=_objects)
+
+            else:
+                _func = getattr(http_name, client._http)
+                for _index, __obj in enumerate(_objects):
+                    if __obj is None:
+                        _id = kwargs.get(kwarg_name)[_index]
+                        _kwargs = kwargs
+                        _kwargs.pop(kwarg_name)
+                        _kwargs[kwarg_name[:-1]] = _id
+                        _request = _func(**_kwargs)
+                        _objects[_index] = _request
+                return _http_request(_obj, http=client._http, request=_objects)
+
+        _obj: Optional[_T] = None
+
+        force_cache = kwargs.pop("force", None) == "cache"
+        force_http = kwargs.pop("force", None) == "http"
+        if not force_http:
+            _obj = _get_cache(obj, client, kwarg_name, **kwargs)
+
+        if force_cache:
+            return _obj
+
+        elif not force_http and _obj:
+            return _return_cache(_obj)
+
+        else:
+            return _http_request(obj=obj, http=client._http, _name=http_name, **kwargs)
+
+    elif len(args) == 1:
+        return _search_iterable(*args, **kwargs)
+
+
+async def _http_request(
+    obj: Type[_T],
+    http: HTTPClient,
+    request: Union[Coroutine, List[Union[_T, Coroutine]], List[Coroutine]] = None,
+    _name: str = None,
+    **kwargs,
+) -> Union[_T, List[_T]]:
+    if not request:
+        if obj in (Role, Emoji):
+            _guild = Guild(**await http.get_guild(kwargs.pop("guild_id")), _client=http)
+            _func = getattr(_guild, _name)
+            return await _func(**kwargs)
+
+        _func = getattr(http, _name)
+        _obj = await _func
+        return obj(**_obj, _client=http)
+
+    if not isinstance(request, list):
+        return obj(**await request, _client=http)
+
+    return [obj(**await req, _client=http) if isawaitable(req) else req for req in request]
```

Comment:
What about asyncio.gather?

```suggestion
    awaitables = []
    dicts = []
    for req in request:
        if isawaitable(req):
            awaitables.append(req)
        else:
            dicts.append(req)

    loop = get_event_loop()
    group = asyncio.gather(*request)
    result = group.run_until_complete(group)

    requests = awaitables + dicts
    return [obj(**req, _client=http) for req in requests]
```

Reply:
we want to keep it async if it is gotten from HTTP, also I *really* don't want to mess with event loops

Reply:
Instead of doing it one after the other, it would do it all at once and return the results in the same order

Reply:
.. ok I'll modify it but no loop stuff lol

Reply:
actually it would not keep it in order so I won't change it (order would not be kept between not coros and coros)

Reply:
You could also wrap the non-awaitables in another thing so that you could still use gather

Reply:
this approach is too complicated imo

Reply:
```suggestion
    awaitables = []
    for req in request:
        if isawaitable(req):
            awaitables.append(req)
        else:
            async def awaitable():
                return req
            awaitables.append(awaitable())

    loop = get_event_loop()
    group = asyncio.gather(*awaitables)
    result = loop.run_until_complete(group)

    return [obj(**res, _client=http) for res in result]
```

Reply:
Alright, will do later

Reply:
Try it out and see if there is a speed difference
If there is a major difference, use the better one
Else use the simpler one

Reply:
This does not work, because `Client` already has a event loop and I can't get it to work without the loop stuff, so we won't use it ig

Reply:
Wait a second, you could just *await* it lol
```py
    async def awaitable(param):
        return param

    awaitables = []
    for req in request:
        if isawaitable(req):
            awaitables.append(req)
        else:
            awaitables.append(awaitable(req))

    result = await asyncio.gather(*awaitables)
    return [obj(**res, _client=http) for res in result]
```